crates/icp-cli/src/operations/token/mint.rs

```diff
@@ -0,0 +1,198 @@
+            .to_u64()
+            .ok_or(MintCyclesError::IcpAmountOverflow)?
+    } else {
+        return Err(MintCyclesError::NoAmountSpecified);
+    };
+
+    // Prepare transfer to CMC
+    let account_id = AccountIdentifier::new(
+        &CYCLES_MINTING_CANISTER_PRINCIPAL,
+        &Subaccount::from(user_principal),
+    );
+    let memo = Memo(MEMO_MINT_CYCLES);
+    let transfer_args = TransferArgs {
+        memo,
+        amount: Tokens::from_e8s(icp_e8s_to_deposit),
+        fee: Tokens::from_e8s(ICP_LEDGER_BLOCK_FEE_E8S),
+        from_subaccount: None,
+        to: account_id,
+        created_at_time: None,
+    };
+
+    // Execute ICP transfer
+    let transfer_result = agent
+        .update(&ICP_LEDGER_PRINCIPAL, "transfer")
+        .with_arg(Encode!(&transfer_args).expect("Failed to encode transfer args"))
+        .call_and_wait()
+        .await
+        .context(TransferIcpSnafu)?;
+
+    let transfer_response =
+        Decode!(&transfer_result, TransferResult).expect("ICP ledger transfer result type changed");
+
+    // Handle transfer result
+    let block_index = match transfer_response {
+        Ok(block_index) => block_index,
+        Err(err) => match err {
+            TransferError::TxDuplicate { duplicate_of } => duplicate_of,
+            TransferError::InsufficientFunds { balance } => {
+                let required =
+                    BigDecimal::new((icp_e8s_to_deposit + ICP_LEDGER_BLOCK_FEE_E8S).into(), 8);
+                let available = BigDecimal::new(balance.e8s().into(), 8);
+                return Err(MintCyclesError::InsufficientFunds {
+                    required,
+                    available,
+                });
+            }
+            err => {
+                return Err(MintCyclesError::TransferFailed {
+                    message: format!("{:?}", err),
+                });
+            }
+        },
+    };
+
+    // Notify CMC to mint cycles
+    let notify_response = agent
+        .update(&CYCLES_MINTING_CANISTER_PRINCIPAL, "notify_mint_cycles")
+        .with_arg(
+            Encode!(&NotifyMintArgs {
+                block_index,
+                deposit_memo: None,
+                to_subaccount: None,
+            })
+            .expect("Failed to encode notify mint cycles args"),
+        )
+        .call_and_wait()
+        .await
+        .context(NotifyMintSnafu)?;
+
+    let notify_response = Decode!(&notify_response, NotifyMintResponse)
+        .expect("Notify mint cycles response type changed");
+
+    // Handle notify result
+    let minted = match notify_response {
+        NotifyMintResponse::Ok(ok) => ok,
+        NotifyMintResponse::Err(err) => {
+            return Err(MintCyclesError::NotifyMintFailed {
+                message: format!("{:?}", err),
+            });
+        }
+    };
+
+    // Calculate display values in TCYCLES (12 decimals)
+    let deposited = BigDecimal::new((minted.minted - CYCLES_LEDGER_BLOCK_FEE).into(), 12);
```

Comment:
```suggestion
    let deposited = BigDecimal::new((minted.minted - CYCLES_LEDGER_BLOCK_FEE).into(), CYCLES_LEDGER_DECIMALS);
```
Would probably be nicer to have this defined next to `CYCLES_LEDGER_BLOCK_FEE`